packages/wallets/src/components/SentEmailContent/SentEmailContent.tsx

```diff
@@ -18,92 +15,38 @@ type SentEmailContentProps = {
 };
 
 const SentEmailContent: FC<SentEmailContentProps> = ({ description, isInvestorPassword = false, platform }) => {
-    const [shouldShowResendEmailReasons, setShouldShowResendEmailReasons] = useState(false);
-    const [hasCountdownStarted, setHasCountdownStarted] = useState(false);
     const { data } = useSettings();
     const { mutate: verifyEmail } = useVerifyEmail();
-    const { isMobile } = useDevice();
     const mt5Platform = PlatformDetails.mt5.platform;
-    const { title } = PlatformDetails[platform ?? mt5Platform];
-    const titleSize = 'md';
-    const descriptionSize = 'sm';
-    const emailLinkSize = isMobile ? 'lg' : 'md';
-    const [count, { resetCountdown, startCountdown }] = useCountdown({
-        countStart: 60,
-        intervalMs: 1000,
-    });
-
-    useEffect(() => {
-        if (count === 0) setHasCountdownStarted(false);
-    }, [count]);
+    const { isMobile } = useDevice();
 
     const { data: activeWallet } = useActiveWalletAccount();
 
     const mt5ResetType = isInvestorPassword
         ? 'trading_platform_investor_password_reset'
         : 'trading_platform_mt5_password_reset';
 
+    useEffect(() => {
+        if (data?.email) {
+            verifyEmail({
+                type: platform === mt5Platform ? mt5ResetType : 'trading_platform_dxtrade_password_reset',
+                url_parameters: {
+                    redirect_to: platformPasswordResetRedirectLink(platform ?? mt5Platform, activeWallet?.is_virtual),
+                },
+                verify_email: data?.email,
+            });
+        }
+    }, [data.email, activeWallet?.is_virtual, mt5Platform, mt5ResetType]);
```

Comment:
so what exactly does trigger the verifyEmail(...)? is it user click?

would it make sense to bind it to click even instead of useEffect? (way easier to control, way less risk of race hazards and accidental execution due to some unrleated changes in data/dependencies e.g. due to react-query doing its magic in background)

Reply:
It triggers on mount, previous design has a button for this but now there's no such button.
Previously it also has a countdown timer to disable the button to prevent users from re-sending the emails, but now it will re-send the email every time the modal is re-mounted

Reply:
I can add a lock here, something like `hasSentEmail` so this only triggers once

Reply:
I'm wondering which component is responsible for actually sending email. 
This component seems to be confirmation, so it should be shown once email is sent - right now we display confirmation and start the process of sending at the same time.

Ideally, "verifyEmail" should be triggered directly by user action, e.g. user click. We should run away as far as possible from "observing" patterns where we do observe some variables and trigger logic based on it, as its really prone to errors/race hazards, its basically the same pattern as mobx.